xarray/core/groupby.py

```diff
@@ -279,7 +279,7 @@ def _yield_binary_applied(self, func, other):
                 raise TypeError('GroupBy objects only support binary ops '
                                 'when the other argument is a Dataset or '
                                 'DataArray')
-            except KeyError:
+            except (KeyError, ValueError):
```

Comment:
An `IndexingError` would be indeed more explicit. Maybe add this is in all the right places in another PR?